src/udiskslinuxprovider.c

```diff
@@ -355,13 +366,10 @@ on_etc_udisks2_dir_monitor_changed (GFileMonitor     *monitor,
       event_type == G_FILE_MONITOR_EVENT_CHANGES_DONE_HINT)
     {
       gchar *filename = g_file_get_basename (file);
+      gchar *id = dup_id_from_config_name (filename);
       if (g_str_has_suffix (filename, ".conf"))
```

Comment:
> This seems to be testing for ".conf" after it tries to subtract 5 from the length of the string?
> 
> Maybe these two hunks should be reverted.

I don't think it removes anything. It uses ``filename`` to get ``id`` and then uses the original ``filename`` for comparison. Or am I missing something?